### ISSUE TITLE
The function oe_get_target_info cannot handle the evidence wrapped by the header that contains format id

### ISSUE THREAD

**Issue by yufjiang** [openenclave/openenclave#3322]
The function oe_get_target_info cannot handle the evidence wrapped by the header that contains format id.
Inside its implemenation, https://github.com/openenclave/openenclave/blob/ce8c48c8ea25bbdef421c30921d5f1271fcaed29/common/sgx/report.c#L163
It assumes that a report, which is not wrapped by the header that contains format id, is passed in. 

Because of that, the local attestation will fail if we set the flag OE_EVIDENCE_FLAGS_EMBED_FORMAT_ID when we call oe_get_evidence, as a wrapped evidence will be returned and then will be immediately pass into the function oe_get_target_info which will not properly offset that evidence data block.

In a nutshell, we currently can generate two different kinds of evidence for local attestation, but only can get the target from one of two evidence formats, which causes the gap in the local attestation.

**Comment by yentsanglee** [comment]
Public API oe_get_target_info() is SGX specific and the first parameter must be oe_report that is returned from oe_get_report(). oe_get_target_info() is not to be mixed with oe_evidence API sets.
